test/rubygems/test_gem_commands_setup_command.rb

```diff
@@ -143,6 +143,22 @@ def test_install_default_bundler_gem
     assert_path_exists 'default/gems/bundler-audit-1.0.0'
   end if Gem::USE_BUNDLER_FOR_GEMDEPS
 
+  def test_install_default_bundler_gem_with_destdir
+    Gem.instance_variable_set :@default_dir, nil
```

Comment:
I'm not fan of this code, but in `Gem::TestCase.setup` we have the
following:

    @default_dir = File.join @tempdir, 'default'
    Gem.instance_variable_set :@default_dir, @default_dir

I understand why it's done this way and don't think we should change
it. This makes testing `--destdir` complicated, and unless there is a
nicer solution, I concluded it might be acceptable to resort to using
`Gem.instance_variable_set` and use `@tempdir` as the value for
`--destdir` option.